app/controllers/admin/images_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Admin::ImagesController < Admin::BaseController
+  before_action :find_edition
+  before_action :enforce_permissions!
+  before_action :build_blank_image
+  layout "design_system"
+
+  def index; end
+
+private
+
+  def find_edition
+    edition = Edition.find(params[:edition_id])
+    @edition = LocalisedModel.new(edition, edition.primary_locale)
+  end
+
+  def build_blank_image
+    if @edition.allows_image_attachments? && @edition.images.none?(&:new_record?)
+      image = @edition.images.build
+      image.build_image_data
+    end
+  end
+
+  def enforce_permissions!
+    unless current_user.can_preview_images_update?
+      redirect_to edit_admin_edition_path(@edition)
```

Comment:
I think it would be good to move this into a separate before action which lives at the top of the before_actions.

Something like

```
before_action :find_edition
before_action :redirect_unless_user_can_preview_image_update
before_action :enforce_permissions!
before_action :build_blank_image
layout "design_system"


def redirect_unless_user_can_preview_image_update
  redirect_to edit_admin_edition_path(@edition) unless current_user.can_preview_images_update?
end
```